building/build-debs/homeworld-keysystem/src/keygen/generate.go

```diff
@@ -0,0 +1,108 @@
+package keygen
+
+import (
+	"keyserver/config"
+	"os"
+	"crypto/rsa"
+	"crypto/rand"
+	"encoding/pem"
+	"crypto/x509"
+	"io/ioutil"
+	"path"
+	"math/big"
+	"crypto/x509/pkix"
+	"time"
+	"golang.org/x/crypto/ssh"
+	"errors"
+)
+
+const AUTHORITY_BITS = 4096
+
+func GenerateTLSSelfSignedCert(key *rsa.PrivateKey, name string, present_as []string) ([]byte, error) {
```

Comment:
Should we factor out a single TLS cert generation function and use it for both the self-signed certs and the certs created by the keyserver?

Reply:
Proooobably.

Reply:
I've looked at this a bit, and it seems like these implementations don't actually reconcile very well -- I'm pulling out a few things, like the serial number generation code, but most of it doesn't get shorter or clearer by factoring.